src/hooks/useRemoteImage.ts

```diff
@@ -0,0 +1,16 @@
+import { useEffect, useState } from 'react'
+
+import * as RD from '@devexperts/remote-data-ts'
+
+export const useRemoteImage = (src: string) => {
+  const [remoteImage, setRemoteImage] = useState<RD.RemoteData<string, string>>(RD.pending)
+
+  useEffect(() => {
+    const image = new Image()
+    image.onload = () => setRemoteImage(RD.success(src))
+    image.onerror = () => setRemoteImage(RD.failure(src))
+    image.src = src
+  }, [src])
```

Comment:
What do you think about to clean-up everything like this?

```
  useEffect(() => {

    const onLoad = () => setRemoteImage(RD.success(src))
    const onError = () => setRemoteImage(RD.failure(src))

    const image = new Image()   
    image.addEventListener('load', onLoad)
    image.addEventListener('error', onError)
    image.src = src

   // clean-up callbacks
    return () => {
       image.removeEventListener('load', onLoad)
       image.removeEventListener('error', onError)
    }
  }, [src])

Reply:
there is no closures here and image object will cleaned from memory the same time as Unmount. with `image.onload` and `image.onerror` properties. so i believe it's not necessary to use listeners here